arango/resolver.py

```diff
@@ -7,20 +7,24 @@
 
 import random
 from abc import ABC, abstractmethod
+from typing import Optional, Set
 
 
 class HostResolver(ABC):  # pragma: no cover
     """Abstract base class for host resolvers."""
 
+    def __init__(self) -> None:
+        self._max_tries: int = 3
+
     @abstractmethod
-    def get_host_index(self) -> int:
+    def get_host_index(self, prev_host_index: Optional[int] = None) -> int:
```

Comment:
To keep the `*Resolver` classes as stateless as possible, I recommend passing in a `Set[int]` here (e.g. "indexes_to_filter") instead of `prev_host_index`. And you can maintain the `Set[int]` in `process_response`.

Reply:
Makes sense. I've addressed what I think you're looking for, we maintain the `Set[int]` in `process_response`: https://github.com/ArangoDB-Community/python-arango/pull/184/files#diff-ff65f7eb9f247417f86b688f08c1dc65226bfe8298866e685349f3a902c062bdR144-R146